### PR TITLE
refactor(g-code-testing): Refactor G-Code testing to store results locally

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

* Move G-Code Parser results files from S3 to being stored in this repo
* Add temporary fix for failing HTTP G-Code Tests
* Updated the Github Action filters

# Changelog

- Store results files inside `comparison_files`
- Modify `SharedFunctionsMixin` to use local files instead of s3
- Add `requested_versions=2` to all module http tests (This is a temporary workaround and needs to be fixed)

# Review requests

None

# Risk assessment

Low

### CODE REVIEW

File: g-code-testing/g_code_test_data/g_code_configuration.py [hunk 1]
```diff
@@ -22,33 +22,20 @@ def get_configuration_path(self) -> str:
...
-        g_code_object = s3.Object(BUCKET_NAME, file_name)
-        result = g_code_object.put(Body=self.execute())
-
-        res = result.get('ResponseMetadata')
-
-        if res.get('HTTPStatusCode') == 200:
-            print(f'{file_name} Uploaded Successfully')
-        else:
-            print(f'{file_name} Not Uploaded')
+        file_path = os.path.join(MASTER_FILES_FOLDER_PATH, self.get_master_file_path())
```
Comment (sfoster1) [inline g-code-testing/g_code_test_data/g_code_configuration.py]:
```python
with open(os.path.join(MASTER_FILES_FOLDER_PATH, self.get_master_file_path()), 'w') as master_file:
    master_file.write(self.execute())
```
?

File: g-code-testing/g_code_test_data/g_code_configuration.py [hunk 2]
```diff
@@ -22,33 +22,20 @@ def get_configuration_path(self) -> str:
...
+        file = open(file_path, "w")
+        file.write(self.self.execute())
```
Comment (sfoster1) [inline g-code-testing/g_code_test_data/g_code_configuration.py]: `self.self`?

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Nits: typos in some of the names - they're `master_comparision_files`, but there's an extra i, should be `master_comparison_files` for the dirname, variable constants, and test file please.

Couple tiny python changes we could make, but overall this looks good and it'll be a lot easier to run locally now, nice!

**Comment by DerekMaggio** [comment]
I also update the Github Action filters for the project to match the other projects in the monorepo

File: g-code-testing/README.md [hunk 3]
```diff
@@ -110,250 +110,15 @@ These tests cover the functionality of all modules
...
 
-### G-Code Parser
+Use the various Make targets in the `Makefile` to run tests using pytest
 
-Using the `GCodeEngine` you can run a Python protocol file against emulation.
-The `GCodeEngine` instance will return a `GCodeProgram` instance
+## Uploading new comparison files
 
-**Example:**
+To update comparison files override files in `g-code-testing/g_code_test_data/comparison_comparision_files` with the output of
```
Comment (sfoster1) [inline g-code-testing/README.md]:
```suggestion
To update comparison files override files in `g-code-testing/g_code_test_data/comparison_comparison_files` with the output of
```

File: g-code-testing/README.md [hunk 4]
```diff
@@ -110,250 +110,15 @@ These tests cover the functionality of all modules
...
-"""
-Using default settings, run protocol at /my/absolute/path/to/my/protocol.py
-against emulation.
-In "Concise" format store parsed G-Code to /where/I/want/to/store/my/output.txt
-"""
-import os.path
-from g_code_parsing.g_code_engine import HTTPGCodeEngine
-from g_code_parsing.g_code_program.supported_text_modes import SupportedTextModes
-from opentrons.hardware_control.emulation.settings import Settings
+You can do this by using `make update-g-code-configuration-comparision name=<comparision_name>`
```
Comment (sfoster1) [inline g-code-testing/README.md]:
```suggestion
You can do this by using `make update-g-code-configuration-comparison name=<comparison_name>`
```

File: g-code-testing/Makefile [hunk 5]
```diff
@@ -64,4 +64,25 @@ lint:
...
+.PHONY: get-g-code-configurations
+get-g-code-configurations:
+	$(pipenv) run python cli.py configurations
+
+.PHONY: run-g-code-configuration
+run-g-code-configuration:
+	$(if $(name),,$(error name variable required))
+	$(pipenv) run python cli.py run ${name}
+
+.PHONY: load-g-code-configuration-comparision
```
Comment (sfoster1) [inline g-code-testing/Makefile]:
```suggestion
.PHONY: load-g-code-configuration-comparison
```

File: g-code-testing/Makefile [hunk 6]
```diff
@@ -64,4 +64,25 @@ lint:
...
+load-g-code-configuration-comparision:
```
Comment (sfoster1) [inline g-code-testing/Makefile]:
```suggestion
load-g-code-configuration-comparison:
```

File: g-code-testing/Makefile [hunk 7]
```diff
@@ -64,4 +64,25 @@ lint:
...
+	$(if $(name),,$(error name variable required))
+	$(pipenv) run python cli.py load-comparison ${name}
+
+
+.PHONY: update-g-code-configuration-comparision
```
Comment (sfoster1) [inline g-code-testing/Makefile]:
```suggestion
.PHONY: update-g-code-configuration-comparison
```

File: g-code-testing/Makefile [hunk 8]
```diff
@@ -64,4 +64,25 @@ lint:
...
+update-g-code-configuration-comparision:
```
Comment (sfoster1) [inline g-code-testing/Makefile]:
```suggestion
update-g-code-configuration-comparison:
```

File: g-code-testing/tests/test_g_code_comparision.py [hunk 9]
```diff
@@ -11,10 +11,8 @@
     [conf.generate_pytest_param() for conf in HTTP_CONFIGURATIONS],
```
Comment (sfoster1) [inline g-code-testing/tests/test_g_code_comparision.py:11]: Please change the name of this file to `test_g_code_comparison.py` (one fewer i)

File: g-code-testing/README.md [hunk 10]
```diff
@@ -110,250 +110,15 @@ These tests cover the functionality of all modules
...
 
-## Python Framework
+## Running tests
 
-### G-Code Parser
+Use the various Make targets in the `Makefile` to run tests using pytest
 
-Using the `GCodeEngine` you can run a Python protocol file against emulation.
-The `GCodeEngine` instance will return a `GCodeProgram` instance
+## Uploading new comparison files
```
Comment (sfoster1) [inline g-code-testing/README.md]:
```suggestion
## Saving new comparison files
```

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Couple more typos (inline comments) but the code looks good

**Review by sfoster1 — APPROVED** [review]
nice!

**Pull request merged**